src/signaling.ts

```diff
@@ -65,10 +65,17 @@ class Signaling extends EventEmitter {
     }
   }
 
-  requestToPublish(mediaTypes: MediaType[]): Promise<SdpRequest> {
-    return this.ws?.call("requestToPublish", {
+  requestToPublish(mediaTypes: MediaType[], alias ?: string): Promise<SdpRequest> {
+    let params: object = {
       mediaTypes: mediaTypes,
-    }) as Promise<SdpRequest>;
+    };
+    if (alias) {
+      params = {
+        mediaTypes: mediaTypes,
+        alias: alias,
+      }
+    }
```

Comment:
you are setting mediaTypes twice, better would be:
```
if (alias) {
  params.alias = alias
}
```

Reply:
Typscripting hated modifying an already constructed object and threw errors attempting to compile.

Reply:
ok, in that case, can you do:
```
if (alias) {
      params = {
        mediaTypes: mediaTypes,
        alias: alias,
      }
} else {
    params = {
        mediaTypes: mediaTypes
    }
}
```

Reply:
Done